petastorm/fs_utils.py

```diff
@@ -183,3 +183,15 @@ def __getstate__(self):
         raise RuntimeError('Pickling FilesystemResolver is not supported as it may contain some '
                            'a file-system instance objects that do not support pickling but do not have '
                            'anti-pickling protection')
+
+
+def get_filesystem_and_path_or_paths(url_or_urls, hdfs_driver):
+    if isinstance(url_or_urls, list):
+        fs = FilesystemResolver(url_or_urls[0], hdfs_driver=hdfs_driver).filesystem()
+        path_or_paths = [FilesystemResolver(url, hdfs_driver=hdfs_driver).get_dataset_path()
```

Comment:
Can you please confirm that this does not end up being very expensive, especially in the case of `libhdfs` driver? (Recreating libhdfs filesystem, say 1000 times). 
Not too found of how the `FilesystemResolver` is design right now, perhaps an easy fix is to add an url-string only implementation of `get_dataset_path` as a standalone function, just based on the `urlparse().scheme`?

Can we easily raise an exception if the FS of all urls is not the same?